src/property.hpp

```diff
@@ -51,6 +53,9 @@ enum class PropertyType : unsigned char {
     // deprecated and remains only for reading old files
     Any    = 9,
 
+    ObjectId = 10,
+    Decimal = 11,
```

Comment:
Why not `Decimal128`?

Reply:
We don't have Int64 and Int32.

Reply:
So why does the PR's title contain `Decimal128`?

Reply:
As far as I can tell, according to mongo's type [documentation](https://docs.mongodb.com/manual/reference/bson-types/#document-bson-type-date) the standard is to use `Decimal128` for enum type names, but use "decimal" for the string type description name. I can change the property type enum according to that.